eth/vm/chain_context.py

```diff
@@ -0,0 +1,25 @@
+from typing import Optional
+
+from eth.abc import ChainContextAPI
+from eth.validation import (
+    validate_uint256,
+)
+
+
+class ChainContext(ChainContextAPI):
+    """
+    This immutable object houses chain information that remains constant for the entire context of
+    the VM execution.
+    """
+    __slots__ = ['_chain_id']
+
+    def __init__(self, chain_id: Optional[int]) -> None:
+
+        if chain_id is None:
+            chain_id = 0  # Default value (invalid for public networks)
+        validate_uint256(chain_id)
```

Comment:
Note: update this to ensure `uint64`, which is the suggested maximum due to the mathematical configuration of EIP-155